src/Compilers/Core/Portable/MemberDescriptor.cs

```diff
@@ -114,6 +114,27 @@ internal static ImmutableArray<MemberDescriptor> InitializeFromStream(Stream str
             return builder.ToImmutable();
         }
 
+        /// <summary>
+        /// The type Id may be:
+        ///     (1) encoded in a single byte (for types below 254)
+        ///     (2) encoded in two bytes (255 + extension byte) for types below 512
+        /// </summary>
+        private static short ReadTypeId(Stream stream)
+        {
+            var firstByte = (byte)stream.ReadByte();
+
+            if (firstByte == (byte)WellKnownType.ExtendedType)
+            {
+                short result = (short)stream.ReadByte();
+                result += (short)WellKnownType.ExtendedType;
+                return result;
```

Comment:
@AlekseyTs, I cannot add two bytes (it would overflow) or add two shorts with the `+` operator. That's why I used this longer way.